de-lib/src/main/java/org/iplantc/de/tools/client/presenter/ManageToolsViewPresenter.java

```diff
@@ -308,16 +310,28 @@ public HandlerRegistration addSelectionChangedHandler(SelectionChangedEvent.Sele
 
     @Override
     public void onShowToolInfo(ShowToolInfoEvent event) {
-       toolInfoDialogProvider.get(new AsyncCallback<ToolInfoDialog>() {
-           @Override
-           public void onFailure(Throwable throwable) {
+        dcService.getAppsForTool(event.getTool().getId(), new AppsCallback<List<App>>() {
+
+            @Override
+            public void onFailure(Integer statusCode, Throwable exception) {
```

Comment:
What do you mean ? - An error with the service ? or there are no apps using the selected tool? The later case is handled.